nixos/modules/system/boot/systemd-nspawn.nix

```diff
@@ -87,16 +194,30 @@ let
     let base = {
       text = ''
         [Exec]
+        ${let env = def.environment;
+          in concatMapStrings (n:
+            let s = optionalString (env."${n}" != null)
+              "Environment=${builtins.toJSON "${n}=${env.${n}}"}\n";
```

Comment:
I guess that at least some of those key-value generators could be simplified by using `lib.generators.toKeyValue`.

Reply:
`toKeyValue` has a bit different behavior... But I could find simplified form in other code locations. Thanks!

Reply:
I see :+1: